### PR TITLE
Change `once`, `twice`, and `never` properties to methods

### PR DESCRIPTION
**Pull request by ollipa**
Changed `once`, `twice`, `never` and `mock` properties to methods. This allows code completion to work for these methods. Previously it was possible to call these methods without parentheses as if they were properties.

To avoid silently breaking user code (eg. `mock.once` would get ignored), an exception is raised if one of these methods is called without parentheses.

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -33,11 +34,13 @@ Removed
...
 - Drop Pytest 4.x support.
 - Remove unittest2 and nose integrations. unittest2 and nose are not maintained anymore.
+- **BREAKING CHANGE**: Removed support for calling `once`, `twice`, `never`, and `mock` methods
+  without parentheses. This allows code completion and static analysis to work with these methods.
 
 Fixed
 #####
 
-- Fix ``with_args`` not working built-in functions and methods.
+- Fix `and_raise` allows invalid arguments for an exception.
```
Comment (christophe-riolo) [inline CHANGELOG.rst:44]: Why did you change this line ?

File: src/flexmock/_api.py [hunk 2]
```diff
@@ -562,32 +565,20 @@ def __init__(
         self._verified = False
         self._callable = True
         self._local_override = False
+        # Remove in 1.0
```
Comment (christophe-riolo) [inline src/flexmock/_api.py]: Can you make an issue for release 1.0 and put the #number here, so it's easier to find ? we might end up with a couple more of those.

**Review by christophe-riolo — APPROVED** [review]
LGTM, just a couple of comments but it's minor :+1:

File: CHANGELOG.rst (hunk 1 above)
Comment (ollipa) [reply CHANGELOG.rst:44]: This fix was backported to 0.10.x branch (https://github.com/flexmock/flexmock/blob/master/CHANGELOG.rst#release-0108) so this was a duplicate entry.

File: src/flexmock/_api.py (hunk 2 above)
Comment (ollipa) [reply src/flexmock/_api.py]: Done! Issue #96. I also made a milestone for 1.0 version.
